### ISSUE TITLE
Extract CRISPR location in bacterial genome from CRISPR database

### ISSUE THREAD

**Issue by mbonsma** [goyalsid/phageParser#35]
We want to know where in a bacterial genome a particular CRISPR region is. With this information, the Genbank file for an organism can be searched and key features (such as genes) can be identified near CRISPRs. 

The [CRISPRdb database](http://crispr.u-psud.fr/crispr/) has a utility to display a file with the properties of an individual CRISPR that also contains the start and end location in the genome. Here's an [example](http://crispr.u-psud.fr/crispr/databases/Output/643561/NC_015138/NC_015138_Crispr_1). 

Part 1) I don't know if this is possible, but it would be great if there was a way to download these files in bulk. Anyone with knowledge on interfacing with websites, this is for you!

Part 2) Once all the files are somewhere accessible, we need a script to extract the CRISPR start and end positions from the file and write them to an output file for all organisms. The output file should have a column with the main accession number for the organism (i.e. `NC_015138`), a column with the CRISPR locus identifier (i.e. `1`), a column with the start position, and a column with the end position. 

Here are the relevant lines in the example file:
```
# Sequence: NC_015138 
# Description: Acidovorax avenae subsp. avenae ATCC 19860 chromosome, complete genome.
# Length: 5482170
# Id: NC_015138
#
#=========================================================================
# Crispr Rank in the sequence: 5
# Crispr_begin_position: 1785060		 Crispr_end_position: 1785158
```

**Comment by daisieh** [comment]
@mbonsma, what, in this example, is the "CRISPR locus identifier"?

**Comment by mbonsma** [comment]
@daisieh, good point, it's not obvious - it's the "Crispr Rank in the sequence", so in the above few lines it would be 5.

**Comment by shirishgoyal** [comment]
Can you list steps how we got that link mentioned for "example" ?
Then we can derive a way to crawl page and seek info needed.

**Comment by mbonsma** [comment]
@shirishgoyal - very good. Here's how I got the example file:

- From the [CRISPRdb home page](http://crispr.u-psud.fr/crispr/), click 'Browse CRISPRs; on the left panel.
- Scroll down to the entry 'Acidovorax avenae subsp. avenae ATCC 19860' and click it, which takes you [here](http://crispr.u-psud.fr/cgi-bin/crispr/SpecieProperties.cgi?Taxon_id=643561).
- Select the bubble under the 'Select Sequence' column heading and then click 'Find CRISPRS' on the bottom left, which takes you [here](http://crispr.u-psud.fr/crispr/CRISPRProperties.php?RefSeq=NC_015138&Taxon=643561).
- Select one of the boxes in the table (in the example above it's the first box), then click 'Submit'. 
- Scroll to the very bottom of the next page and click 'Display CRISPR properties file'. 

You could also get to the second last step by clicking on the appropriate accession number (NC_015138_1) on the [CRISPR utitilties page](http://crispr.u-psud.fr/crispr/CRISPRUtilitiesPage.html). There are probably other ways to navigate to the CRISPR Details page too. 

Let me know if that helps.

**Comment by mbonsma** [comment]
Addressed in PR #54.

**Comment by mbonsma** [comment]
Solved with PR #54 - `extract_CRISPRdb.py`
